### PR TITLE
build: Add Dockerfile to rebuild in container

### PR DESCRIPTION
**Pull request by rzr**
This docker file is a conveinance tool for developers,
for production deployment please consider this project:

https://github.com/mozilla-iot/gateway-docker

Also run-app.sh can be used to start gw,
more changes to come later.

Usage:

    docker-compose up

Change-Id: Ia082fdcb1b945645ff5c058760dbefed8843eea8
Signed-off-by: Philippe Coval <philippe.coval@osg.samsung.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/ .
+#}
+# WARNING: this docker file is *ONLY* for developer convenience
+# WARNING: for production deployment please consider supported project:
+# WARNING: https://github.com/mozilla-iot/gateway-docker
+
+FROM debian:stable
+MAINTAINER Philippe Coval (p.coval@samsung.com)
```
Comment (webwurst) [inline Dockerfile]: Just a heads-up: https://docs.docker.com/engine/reference/builder/#maintainer-deprecated

**Comment by mrstegeman** [comment]
@rzr What's the purpose of this since we already have a full production-ready Docker image?

**Comment by rzr** [comment]
it can be used for testing local changes on a fresh env

**Comment by mrstegeman** [comment]
It would probably be useful to allow sharing in the `~/.mozilla-iot` volume.

**Comment by rzr** [comment]
Actually I prefer to not mix previously used contents, I prefer to start on blank container to make sure everything is reproducible on the current codebase, but I can submit a config sharing change in later commit.

**Pull request merged**